### ISSUE TITLE
how to extract unique mapped reads?

### ISSUE THREAD

**Issue by bilibilij** [haowenz/chromap#20]
Hi,
    We have mapped our illumina short reads to our genome using typical commands with --SAM of chromap. 
![45b8ab262cb63858d73611e96de4d93](https://user-images.githubusercontent.com/49976278/127955610-04cd37bd-4bae-4cb0-ad1e-d2a1c492e90e.png)
But I failed to find the tag like "NH:i:<N>" or "XT:A:U" at the 12th line of our sam output.
![0bb284cfc481e89bfb3b134b6b7bae4](https://user-images.githubusercontent.com/49976278/127955649-e43cb659-8d55-482e-aba8-ba5f5415e276.png)

Can I extract unique mapped reads from the sam output of chromap?

**Comment by mourisl** [comment]
For the multi-mapped reads, their mapq will be 0 and will be filtered out in Chromap's output by default (-q 30). So you can assume the alignments in Chromap's output are all uniquely aligned.

**Comment by bilibilij** [comment]
![e91e7c69b9158b088f58a4b3181b40e](https://user-images.githubusercontent.com/49976278/128125880-495c5196-4b1d-4e64-8dd6-fb4ebbfb7d46.png)
So after filtered mapq 30, there are only about 25% reads  be successsfully mapped to the reference genome in the SAM output?

**Comment by mourisl** [comment]
It looks like this is the case. Is your data single-end? Or the genome is very repetitive?

**Comment by bilibilij** [comment]
Thank you for your reply , we are assembling a highly heterozygous  and highly repetitive plant genome and this is the mapping ratio before polish.  We would like to polish the draft genome and to see some improvement.

**Comment by mourisl** [comment]
I think you can try the option -q 1, so Chromap can output more reads.

**Comment by Solyris83** [comment]
I would like to add on to this topic, in particular how the reported unique reads can be extracted fully. I have the run below 
`chromap --preset hic -t 32 -r GRCh38.primary_assembly.genome.fa.gz -x GRCh38.chromap_r151.index -1 OmniC_400M_R1.fastq.gz -2 OmniC_400M_R2.fastq.gz -o OmniC_400M_HIC.pair 2> chromap_400mil_default.log
`
with summary of Chromap found here
`Number of reads: 800000000.
Number of mapped reads: 628370352.
Number of uniquely mapped reads: 563134612.
Number of reads have multi-mappings: 65235740.
Number of candidates: 20912715113.
Number of mappings: 628370352.
Number of uni-mappings: 563134612.
Number of multi-mappings: 65235740.
Sorted and outputed mappings in 432.54s.
# uni-mappings: 307445741, # multi-mappings: 6739328, total: 314185069.
Number of output mappings (passed filters): 263511542
Total time: 2107.27s.
`
I am interested in output of this "uni-mappings: 307445741" reads into a pair file. So according to @mourisl -q 1 would get us the unique reads, which is the default setting in --preset hic but from the output value "Number of output mappings (passed filters): 263511542", it seems like there are other filters active which did not generate the needed # uni-mappings: 307445741, may I know what other filters can I tweak for the required output?

Regards
Solyris

**Comment by haowenz** [comment]
I agree there is some ambiguity in the log. There is no hard cutoff of MAPQ for unique mappings. They are two different things. A unique mapping can get a low quality score like 0 if the mapping is not good (e.g., very different from the aligned region on the reference, or the second best mapping is also very good, which makes the best mapping not that reliable).

**Comment by Solyris83** [comment]
I see, thanks for the prompt reply. Is there absolutely no way to just output uniquely-mapped results then?

I have tried a demo dataset from Omni-C (https://omni-c.readthedocs.io/en/latest/data_sets.html ) with both Chromap (default MAPQ>30) and BWA-mem filtered at MAPQ>40 (as recommended by Dovetail Omni-C for BWA-mem for uniquely mapped reads). BWA was able to get me 75.55% uniquely-mapped reads (302,186,679 / 400,000,000) but in Chromap I am filtering out almost all the reads.
It would be very much appreciated if can you point me to any resource for interpreting the MAPQ generated in Chromap. And what is the reason MAPQ>1 for --prefix hic

**Comment by haowenz** [comment]
There used to be an option to output unique mapping regardless of mapq. But that option was not made available in the release. We might make it available in the future for advanced users like you.

The default mapq cutoff for Chromap on Hi-C data is 1, which is also what is in the preset. This is also the default used in [pairtools](https://github.com/open2c/pairtools), which also uses 1 for BWA. Different aligners (e.g., BWA, Bowtie2, Chromap and etc.) have various methods to compute MAPQ and there is no single best way to do that. This also results in different MAPQ range for different aligners on the same dataset. So for Chromap, you can use 1 as MAPQ cutoff, which is our suggestion.
